### PR TITLE
Parse Journal filters from get params

### LINKED ISSUE #239: Store Journal filters in GET-params
Split from #217, ref #210.

### PR DESCRIPTION
**Pull request by aumayr**
fix #239

### CODE REVIEW

**Comment by aumayr** [comment]
@yagebu Please have a look at the code, and give your 👍 if I can merge.

File: fava/static/package.json [hunk 1]
```diff
@@ -30,6 +30,7 @@
     "jquery-dragster": "github:catmanjan/jquery-dragster",
     "jquery-query-object": "^2.2.3",
     "jquery-stupid-table": "github:medsabir/Stupid-Table-Plugin#feat-npm-publish",
-    "mousetrap": "^1.5.3"
+    "mousetrap": "^1.5.3",
+    "query-string": "^4.1.0"
```
Comment (yagebu) [inline fava/static/package.json]: This adds a second JS dependency to deal with the query string. Do we really need both?

**Comment by yagebu** [comment]
I would propose the following format to store which types to show: Have one query parameter (simply called `show` maybe), that takes a list of types. Something like `[...]?show=open&show=closed[...]`. Makes for shorter and more "human-readable" URLs.

The list should also be complete and not depend on the config. Currently it doesn't give you real "permalinks" as only values that are different from the settings are stored.

File: fava/static/package.json (hunk 1 above)
Comment (aumayr) [inline fava/static/package.json]: Which one do you mean?

**Comment by aumayr** [comment]
> I would propose the following format to store which types to show: Have one query parameter (simply called `show` maybe), that takes a list of types. Something like `[...]?show=open&show=closed[...]`. Makes for shorter and more "human-readable" URLs.

I like this idea, and should be easy to implement. 

> The list should also be complete and not depend on the config. Currently it doesn't give you real "permalinks" as only values that are different from the settings are stored.

In general this is true and a good idea as well. I hesitate to implement this for just this feature - it would be generally better to implement #214 (Move fava settings to source file options) and have all the settings in the beancount-file, as otherwise `fava`-specific settings are spread over two places, when they should be in just one.

What do you think

File: fava/static/package.json (hunk 1 above)
Comment (yagebu) [inline fava/static/package.json]: `jquery-query-object`

Comment (aumayr) [inline fava/static/package.json]: Ahh, got it. Didn't see that, will fix it/remove one of the two. Do you have a preference for which to remove?

Comment (yagebu) [inline fava/static/package.json]: Even the author of `jquery-query-object` [doesn't recommend](https://github.com/alrusdi/jquery-plugin-query-object#disclaimer--) using it, so I'm for keeping `query-string` instead (or considering [URI.js](https://github.com/medialize/URI.js) if we need something more powerful)

**Comment by aumayr** [comment]
@yagebu Implemented in 6a003ae

**Comment by aumayr** [comment]
> The list should also be complete and not depend on the config. Currently it doesn't give you real "permalinks" as only values that are different from the settings are stored.

Any use case for the full list?

**Comment by yagebu** [comment]
> Any use case for the full list?

This feature exists to give you the ability to save a permalink, right? And currently it just doesn't do that. If you change your settings, the behaviour of this link will change, so it's not a permalink.

Also, if I save a link, it would be nice to be able to tell just from the link what it will do. With `?show=open` I would expect to see *just* open directives and not whatever else I've enabled in my settings. There's also no need for `hide=...` then, as everything that's not shown is hidden obviously.

**Comment by aumayr** [comment]
OK, implemented it this way in dac91f8.

**Pull request merged**